src/goose.rs

```diff
@@ -1424,63 +1490,83 @@ impl GooseUser {
-    ///                 // We only need to check pages that returned a success status code.
-    ///                 if response.status().is_success() {
-    ///                     match response.text().await {
-    ///                         Ok(text) => {
-    ///                             // If the expected string doesn't exist, this page load
-    ///                             // was a failure.
-    ///                             if !text.contains("this string must exist") {
-    ///                                 // As this is a named request, pass in the name not the URL
-    ///                                 user.set_failure(&mut goose.request);
-    ///                             }
+    ///     async fn loadtest_index_page(user: &GooseUser) -> GooseTaskResult {
+    ///         let mut goose = user.get_named("/", "index").await?;
+    ///
+    ///         if let Ok(response) = goose.response {
+    ///             // We only need to check pages that returned a success status code.
+    ///             if response.status().is_success() {
+    ///                 match response.text().await {
+    ///                     Ok(text) => {
+    ///                         // If the expected string doesn't exist, this page load
+    ///                         // was a failure.
+    ///                         if !text.contains("this string must exist") {
+    ///                             // As this is a named request, pass in the name not the URL
+    ///                             return user.set_failure("string missing", &mut goose.request, None, None);
     ///                         }
-    ///                         // Empty page, this is a failure.
-    ///                         Err(_) => user.set_failure(&mut goose.request),
+    ///                     }
+    ///                     // Empty page, this is a failure.
+    ///                     Err(_) => {
+    ///                         return user.set_failure("empty page", &mut goose.request, None, None);
     ///                     }
     ///                 }
-    ///             },
-    ///             // Invalid response, this is already a failure.
-    ///             Err(_) => (),
-    ///         }
+    ///             }
+    ///         };
+    ///
+    ///         Ok(())
     ///     }
     /// ````
-    pub fn set_failure(&self, request: &mut GooseRawRequest) {
+    pub fn set_failure(
+        &self,
+        tag: &str,
+        request: &mut GooseRawRequest,
+        headers: Option<&header::HeaderMap>,
+        body: Option<&str>,
+    ) -> GooseTaskResult {
         // Only send update if this was previously a success.
         if request.success {
             request.success = false;
             request.update = true;
             self.send_to_parent(&request);
         }
+        // Log failure, converting `&mut request` to `&request` as needed by `log_debug()`.
+        self.log_debug(&tag, Some(&*request), headers, body);
```

Comment:
Maybe make it possible to log to stdout as well here - as the code did before?

Reply:
I meant to do that but forgot, thanks for catching it! we now use `info!()` so the debug output shows up when `-v` is enabled.